src/testing/deploy/deploy_test.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/v2/src/auth"
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/minikubetestenv"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/internal/testutil"
+)
+
+func TestDeployEnterprise(t *testing.T) {
+	k := testutil.GetKubeClient(t)
+	minikubetestenv.DeleteRelease(t, context.Background(), k) // cleanup cluster
```

Comment:
I know we've had to be defensive with cleanup in the past, but I would prefer that these new APIs don't force that on the tests.  Can we do this DeleteRelease in `InstallReleaseEnterprise` if there is already one there?